app/src/main/java/eu/theinvaded/mastondroid/viewmodel/LoginViewModel.java

```diff
@@ -128,13 +107,23 @@ public void call(Token token) {
                             @Override
                             public void call(Throwable throwable) {
                                 isSignIn.set(!isSignIn.get());
-
-                                viewModel.showLoginError();
+                                viewModel.setError("login_process", "login_failed_message", throwable.toString());
```

Comment:
This looks a bit scary to me. If the resource name is changed in the layout for whatever reason, and it's not reflected here it will crash at runtime.